agent/engine/docker_task_engine.go

```diff
@@ -1138,7 +1143,33 @@ func (engine *DockerTaskEngine) AddTask(task *apitask.Task) {
 		engine.emitTaskEvent(task, err.Error())
 		return
 	}
-
+	if task.IsEBSTaskAttachEnabled() {
+		if engine.ebsCsiDriver != nil {
+			logger.Info("engine ebs CSI driver is running", logger.Fields{
+				field.TaskID: engine.ebsCsiDriver.GetID(),
+			})
+		} else {
+			var csiStartErr error
+			// TODO update 'ebs-csi-driver' as config param or const
+			if ebsCsiDaemonManager, ok := engine.daemonManagers["ebs-csi-driver"]; ok {
+				engine.ebsCsiDriver, csiStartErr = ebsCsiDaemonManager.CreateDaemonTask()
+				engine.AddTask(engine.ebsCsiDriver)
+				logger.Info("docker_task_engine: Added EBS CSI task to engine")
+			} else {
+				csiStartErr = errors.New("Unable to find ebs-csi-driver in engine")
+			}
+			if csiStartErr != nil {
+				logger.Error("Unable to start ebsCsiDaemon for task in the engine", logger.Fields{
+					field.TaskID: task.GetID(),
+					field.Error:  err,
+				})
+				task.SetKnownStatus(apitaskstatus.TaskStopped)
+				task.SetDesiredStatus(apitaskstatus.TaskStopped)
+				engine.emitTaskEvent(task, err.Error())
+				return
+			}
+		}
```

Comment:
nit: I see daemon managers initialization has a common implementation. If this is a common pattern we expect for managed daemons, this part can be abstracted into a common function

Reply:
good point -- I'll pick this up in a future rev if ok?